frontend/public/components/factory/table.tsx

```diff
@@ -124,16 +136,22 @@ const stateToProps = ({UI}, {data = [], defaultSortField = 'metadata.name', defa
 
   if (loaded) {
     let sortBy: string | Function = 'metadata.name';
-    if (currentSortField) {
+    if (currentSortField && !currentSortFunc) {
       // Sort resources by one of their fields as a string
       sortBy = resource => sorts.string(_.get(resource, currentSortField, ''));
-    } else if (currentSortFunc && sorts[currentSortFunc]) {
+    } else if (!currentSortField && currentSortFunc && sorts[currentSortFunc]) {
       // Sort resources by a function in the 'sorts' object
       sortBy = sorts[currentSortFunc];
     }
 
-    // Always set the secondary sort criteria to ascending by name
-    newData = _.orderBy(newData, [sortBy, 'metadata.name'], [currentSortOrder, SortByDirection.asc]);
+    if (currentSortField && currentSortFunc && sorts[currentSortFunc]) {
```

Comment:
this logic should still be OK. I think you are missing a rebase, but essentially the default sort state will get set in the Table `constructor` and find your `currentSortField` passed first, and should set the default sort state correctly still...

i'm good w/ this 👍